File: 二叉树/101.对称二叉树.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        return isFlag(root.left,root.right);
    }
    boolean isFlag(TreeNode left,TreeNode right){
        if(left==null || right==null){
            return left==right;
        }
        return left.val==right.val && isFlag(left.left,right.right) && isFlag(left.right,right.left);
    }
}

File: 二叉树/101.对称二叉树.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
	return isFlag(root.Left, root.Right)
}

func isFlag(left *TreeNode, right *TreeNode) bool {
	if left == nil || right == nil {
		return left == right
	}
	return left.Val == right.Val && isFlag(left.Left, right.Right) && isFlag(left.Right, right.Left)
}